docs/proposals/node-allocatable.md

```diff
@@ -0,0 +1,158 @@
+
+### API changes
+
+#### Allocatable
+
+Add `Allocatable` (4) to
+[`NodeStatus`](https://htmlpreview.github.io/?https://github.com/kubernetes/kubernetes/HEAD/docs/api-reference/v1/definitions.html#_v1_nodestatus):
+
+```
+type NodeStatus struct {
+  ...
+  // Allocatable represents schedulable resources of a node.
+  Allocatable ResourceList `json:"allocatable,omitempty"`
+  ...
+}
+```
+
+Allocatable will be computed by the Kubelet and reported periodically to the API server. It is
+defined to be:
+
+```
+   [Allocatable] = [Machine Capacity] - ( [Reserved] + [System Reserved] + [Kernel usage (smoothed)] )
+```
+
+The scheduler will use `Allocatable` in place of `Capacity` when scheduling pods, and the Kubelet
+will use it when performing admission checks.
+
+- Since kernel usage can fluctuate and is out of kubernetes control, `Allocatable` may vary over
+  time.
+
+#### System Reserved
+
+- **TODO**: Determine which resources *must* be specified (CPU+memory?)
+- **TODO**: Decide on format (parsed string vs. individual resource flags)
+
+`SystemReserved` is a **required** parameter specifying resources reserved for system components
+(4). It is provided as a command-line flag to the Kubelet at startup, and therefore cannot be
+changed during normal Kubelet operation (this could change in the [future](#future-work)).
+
+#### Reserved
+
+`Reserved` is an **optional** parameter specifying reserved resources (2). It is specified as a
+Kubelet command line flag using the same format as [system reserved](#system-reserved).
+
+## Issues
+
+### System reservation is smaller than system component usage
+
+Solution **TBD**
+
+**Option 1:** Do nothing (best effort). Let the system daemons overflow the reserved resources and
+  hope for the best. If the machine usage is less than Allocatable, there will be some room for
+  overflow and the node should continue to function. If the machine has been scheduled to capacity
+  (worst-case scenario) it may enter an unstable state, which is the current behavior in this
+  situation.
+
+**Option 2:** Assume the worst and transition the node to the not-ready state. The kubelet could
+  attempt to reduce usage by evicting pods until usage comes under control. At that point the node
+  could be made available again.
+
+### Version discrepancy
+
+**API server / scheduler is not allocatable-resources aware:** If the Kubelet rejects a Pod but the
+  scheduler expects the Kubelet to accept it, the system could get stuck in an infinite loop
+  scheduling a Pod onto the node only to have Kubelet repeatedly reject it. To avoid this situation,
+  we will do a 2-stage rollout of `Allocatable`. In stage 1 (targeted for 1.2), `Allocatable` will
+  be reported by the Kubelet and the scheduler will be updated to use it, but Kubelet will continue
+  to do admission checks based on `Capacity` (same as today). In stage 2 of the rollout (targeted
+  for 1.3), the Kubelet will start doing admission checks based on `Allocatable`.
```

Comment:
It would be ideal for new clusters to get the desired behavior and benefits right away.  Is it possible for those clusters to jump to stage 2 directly as part of 1.2 in your view?